fossology/fossology
# fix(ojo): Handle dual-license and SPDX new naming

## Description

1. Report Dual-license if SPDX identifier contains or, and or with.
2. Normalize "-or-later" with "+" and "-only" with SPDX old and new naming conventions.

### Changes

1. Added new regex to find dual licenses.
1. Replace `std::basic_string::replace` with `std::basic_string::erase`.

## How to test

1. Upload a test file with SPDX License Identifiers.
1. Check if ojo is reporting "Dual-license" if identifier contains **or, and or with**.
1. Check if ojo is handling licenses with "-or-later" or "-only" at end (new SPDX naming scheme) with old naming scheme.
1. Check if ojo is reporting "Dual-license" for CLI  execution as well.

Closes #1540
